### PR TITLE
Dockerfile for android

### PR DESCRIPTION
**Pull request by Xreki**
Use the following command to build the developing image and cross-compile Paddle:
```
docker build -t paddle-android:dev . -f Dockerfile.android
docker run --rm -v $PWD:/paddle paddle-android:dev
```

### CODE REVIEW

File: Dockerfile.android [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    pip install -U wheel sphinx && \
+    pip install pre-commit
+
+# git credential to skip password typing
+RUN git config --global credential.helper store
+
+# Fix locales to en_US.UTF-8
+RUN localedef -i en_US -f UTF-8 en_US.UTF-8
+
+RUN curl -sSL https://cmake.org/files/v3.2/cmake-3.2.2.tar.gz | tar -xz && \
```
Comment (wangkuiyi) [inline Dockerfile.android]: I noticed that the base image is Ubuntu 16.04. I think we have use `apt` to install a recent enough cmake, like we did in the [main Dockerfile](https://github.com/PaddlePaddle/Paddle/blob/develop/Dockerfile#L31).

File: paddle/math/MathFunctions.cpp [hunk 2]
```diff
@@ -180,7 +180,6 @@ int getri<double>(const CBLAS_ORDER order,
                   const int lda,
                   const int* ipiv) {
   return dynload::PADDLE_DGETRI(order, N, A, lda, ipiv);
-  return 0;
```
Comment (wangkuiyi) [inline paddle/math/MathFunctions.cpp]: I am far from an expert of the `math` package. Just curious if this change is related to Android?

Comment (Xreki) [inline paddle/math/MathFunctions.cpp]: It is not related to Android. I found this redundant statement accidentally when I was doing this work. Should I do this change in another pull request?

Comment (wangkuiyi) [inline paddle/math/MathFunctions.cpp]: Yeah. I think we should, because usually this change requires a different group of reviewers, who are supposed to be familiar with math code, other than Android building.

But if you are sure this change is correct, I don't insist that we must separate it from this PR into a new PR.  You are welcome to make the decision. :-)

File: Dockerfile.android [hunk 3]
```diff
@@ -0,0 +1,42 @@
+FROM ubuntu:16.04
+MAINTAINER PaddlePaddle Authors <paddle-dev@baidu.com>
```
Comment (gangliao) [inline Dockerfile.android]: Can we add this into Travis CI or TeamCity now?

File: paddle/math/MathFunctions.cpp (hunk 2 above)
Comment (Xreki) [inline paddle/math/MathFunctions.cpp]: Thank you. :)

File: Dockerfile.android (hunk 3 above)
Comment (Xreki) [inline Dockerfile.android]: Do we need to discuss it?

File: Dockerfile.android (hunk 1 above)
Comment (Xreki) [inline Dockerfile.android]: Done.

File: Dockerfile.android (hunk 3 above)
Comment (gangliao) [inline Dockerfile.android]: Yeah
